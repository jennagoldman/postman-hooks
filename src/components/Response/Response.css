section {
  height: 600px;
  overflow: scroll;
}

.react-json-view {
  padding: 20px;
}