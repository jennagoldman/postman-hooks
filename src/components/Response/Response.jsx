import React from 'react';
import ReactJson from 'react-json-view';
import styles from './Response.css';
import PropTypes from 'prop-types';

const Response = ({ response }) => (
  <section className={styles.responseSection}>
    <fieldset>
      <legend>RESPONSE</legend>
      {response ? <ReactJson
        src={response} 
        enableClipboard={false}
        iconStyle="circle"
        displayDataTypes={false}
        groupArraysAfterLength={50}
        theme={{
          base00: '#001121',
          base01: '#ec66d6',
          base02: '#fc40e6',
          base03: '#fc40e6',
          base04: '#ec66d6',
          base05: '#ec66d6',
          base06: '#ec66d6',
          base07: '#fff',
          base08: '#fff',
          base09: '#1df5f1',
          base0A: '#f5f1f1',
          base0B: '#fc40e6',
          base0C: '#fc40e6',
          base0D: '#fc40e6',
          base0E: '#ec66d6',
          base0F: '#ec66d6'
        }}
      /> : null }
    </fieldset>
  </section>

);

Response.propTypes = {
  response: PropTypes.array.isRequired
};

export default Response;
