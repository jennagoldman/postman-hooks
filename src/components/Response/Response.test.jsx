import React from 'react';
import { shallow } from 'enzyme';
import Response from './Response.jsx';

describe('Response', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Response response={''} />);
    expect(wrapper).toMatchSnapshot();
  });
});
