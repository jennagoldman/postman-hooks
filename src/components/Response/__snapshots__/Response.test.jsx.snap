// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response matches a snapshot 1`] = `
<section
  className="responseSection"
>
  <fieldset>
    <legend>
      RESPONSE
    </legend>
    Response data will display here upon successful request
  </fieldset>
</section>
`;
